The patch to google/__init__.py is backported from upstream
https://github.com/protocolbuffers/protobuf/commit/8351926380c7cc91aae6df5695c91426e209f958.patch.

The patch to setup.py is original, as upstream has since removed setup.py.

diff --git google/__init__.py google/__init__.py
index 558561412299..b36383a61027 100644
--- google/__init__.py
+++ google/__init__.py
@@ -1,4 +1,3 @@
-try:
-  __import__('pkg_resources').declare_namespace(__name__)
-except ImportError:
-  __path__ = __import__('pkgutil').extend_path(__path__, __name__)
+from pkgutil import extend_path
+
+__path__ = extend_path(__path__, __name__)
diff --git setup.py setup.py
index cbb9a5906ba3..9d3897c60801 100755
--- setup.py
+++ setup.py
@@ -8,7 +8,6 @@ from distutils import util
 import fnmatch
 import glob
 import os
-import pkg_resources
 import re
 import subprocess
 import sys
@@ -270,8 +269,7 @@ if __name__ == '__main__':
     # deployment target of macOS 10.9 or later, or iOS 7 or later.
     if sys.platform == 'darwin':
       mac_target = str(sysconfig.get_config_var('MACOSX_DEPLOYMENT_TARGET'))
-      if mac_target and (pkg_resources.parse_version(mac_target) <
-                         pkg_resources.parse_version('10.9.0')):
+      if mac_target and (mac_target.split('.') < '10.9.0'.split('.')):
         os.environ['MACOSX_DEPLOYMENT_TARGET'] = '10.9'
         os.environ['_PYTHON_HOST_PLATFORM'] = re.sub(
             r'macosx-[0-9]+\.[0-9]+-(.+)', r'macosx-10.9-\1',
